File: web/src/layout/MultipleDemoXml.ts
//
// MultipleDemoXml.ts
// Created by Butterfly XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'butterfly/dist/views/html'
import { customViewSetDelegate } from 'butterfly/dist/views/CustomView.actual'
import htmlForDefault from './multiple_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.MultipleDemoXml
export class MultipleDemoXml {
    xmlRoot!: HTMLElement;
    list!: HTMLDivElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.list = getViewById<HTMLDivElement>(view, "list");
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/MultipleDemoXml.kt
//
// MultipleDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class MultipleDemoXml {

    lateinit var list: androidx.recyclerview.widget.RecyclerView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.multiple_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        list = view.findViewById<androidx.recyclerview.widget.RecyclerView>(R.id.list)
        
        
        
        return view
    }
}